### ISSUE TITLE
BottomVision: failure in vision Pipeline when running nozzle-tip calibration returns strange error message

### ISSUE THREAD

**Issue by ldpgh** [openpnp/openpnp#798]
## Actual Result
Calibration stops with strange message
![image](https://user-images.githubusercontent.com/13517069/50725849-beca9680-1104-11e9-9ecf-46801b692a9f.png)

## Root Cause
If the BottomVision Pipeline during Nozzle-tip calibration does not return a valid measurement the method `ReferenceNozzleTip.java:findCircle()` does not return anything.

The exception is thrown when accessing an element of results
```
...
                Object result = pipeline.getResult(VisionUtils.PIPELINE_RESULTS_NAME).model;
Logger.debug("((List) result).size() = {}", ((List) result).size());
                if (result instanceof List) {
                    if (((List) result).get(0) instanceof Result.Circle) {
...
```
 ((List) result).size()) = 0 in case of an issue.

There is no exception handler ...just try ... is implemented.  It returns from the function without any results, because no regular return is called.

For testing .... in order to have no_valid measurement change MaskCircle:diameter=1

The exception is elevated back and creates an error message ... see screenshot above.

Debugging the exception (step by step in the debugger) ends in FutureTask (?) ... see screenshot.
![image](https://user-images.githubusercontent.com/13517069/50725887-765fa880-1105-11e9-9b5e-9631baf3d225.png)

## Expected Result
Extend  `ReferenceNozzleTip.java:findCircle()`  with `finally{}` and return more information about the root cause.

**Comment by vonnieda** [comment]
Note that this is an unreleased, unfinished feature and bugs are expected. As noted in https://github.com/openpnp/openpnp/issues/235 - someone will need to take this feature over and complete it before it can be used. I think it would be better to keep the discussion and issues on this feature in https://github.com/openpnp/openpnp/issues/235 until it is available to users.

**Comment by ldpgh** [comment]
Note for me:
- If "results" stage is not implemented in theCV-pipeline it immediately jumps down to `return location`. A `catch (Excpetion ee) { ... }` is ignored ... unkwown why. Removed for now.
- If "results" stage does not return valid date the size of the list `result` is 0.

**Comment by ldpgh** [comment]
fix is in PR #798
